contrib/twitchtv/twirp/twirp_test.go

```diff
@@ -0,0 +1,324 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package twirp
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"testing"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/globalconfig"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/twitchtv/twirp"
+	"github.com/twitchtv/twirp/ctxsetters"
+	"github.com/twitchtv/twirp/example"
+)
+
+type mockClient struct {
+	code int
+	err  error
+}
+
+func (mc *mockClient) Do(req *http.Request) (*http.Response, error) {
+	if mc.err != nil {
+		return nil, mc.err
+	}
+	req.Body = nil
```

Comment:
Can you please explain why?

Reply:
https://golang.org/pkg/net/http/#Response
```
    // Request is the request that was sent to obtain this Response.
    // Request's Body is nil (having already been consumed).
    // This is only populated for Client requests.
    Request *Request
```

Reply:
I would appreciate an explanation though. I'm not smart enough to figure it out from your link.

Reply:
Within the scope of a mock, it's not terribly important.
But, based on the docs, a `http.Response` for a client should contain the `Request` it was given but with a nil body. So I did that in the `mockClient`.

Reply:
Added the comment.